### PR TITLE
fix: string date format validation

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by ivan-tymoshenko** [comment]
Please make sure that this test works as expected.
https://github.com/fastify/fast-json-stringify/blob/e083ca6778b6cfeb09be24dd15bae5c6c99791ee/test/date.test.js#L492

File: serializer.js [hunk 1]
```diff
@@ -69,36 +69,36 @@ module.exports = class Serializer {
...
-  asDatetime (date) {
-    const quotes = '"'
+  asDateTime (date) {
+    if (date === null) return '""'
     if (date instanceof Date) {
-      return quotes + date.toISOString() + quotes
+      return '"' + date.toISOString() + '"'
     }
-    return this.asString(date)
+    throw new Error(`The value "${date}" cannot be converted to a date-time.`)
```
Comment (climba03003) [inline serializer.js:77]: What if I passed a string that fulfill `date-time`?
It seems like it only accept `Date` object after this change.

Comment (ivan-tymoshenko) [reply serializer.js:77]: If you pass the string it will be stringified in the `asString` method.

**Review by climba03003 — APPROVED** [review]
LGTM.

File: ajv.js [hunk 2]
```diff
@@ -7,29 +7,22 @@ const ajvFormats = require('ajv-formats')
...
-        return validateDateTimeFormat(date)
-      }
-      if (schema === 'date') {
-        return validateDateFormat(date)
-      }
-      if (schema === 'time') {
-        return validateTimeFormat(date)
-      }
-      return false
+    keyword: 'fjs_type',
```
Comment (mcollina) [inline ajv.js:21]:
```suggestion
    keyword: 'fjs_date_type',
```


I think we should try not to change this.

Comment (ivan-tymoshenko) [reply ajv.js:21]: @mcollina I agree, but it is a completely different meaning of this property now. It's not a date format anymore. It describes how ajv should validate custom type at all. For example, for string type, there would be not only the Date object validation but also Regexp objects and some other exceptions.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Eomm — APPROVED** [review]

**Pull request merged**